File: backend/Organetto/Organetto.UseCases/Boards/Columns/Cards/Data/CardDetailDto.cs
namespace Organetto.UseCases.Boards.Columns.Cards.Data
{
    public record CardDetailDto(long Id, string Title, string Description, long Position, DueDateDto[] DueDates, AttachmentDto[] Attachments)
    {
        public CardDetailDto() : this(0, string.Empty, string.Empty, 0, Array.Empty<DueDateDto>(), Array.Empty<AttachmentDto>())
        {
        }
    }
}

File: frontend/src/dataAccess/cards/models/CardDetailDto.ts
import type { AttachmentDto } from './Attachment'

interface CardDetailDto {
  id: number;                     // long in C#
  title: string;
  description: string | null;
  position: number;
  dueDates: DueDateDto[];
  attachments: AttachmentDto[];
}

interface DueDateDto {
  dueAt: string;                   // ISO datetime string
  isComplete: boolean;
}

interface CardDetailPayload {
  columnId: number,
  id: number,
}

export type { CardDetailDto, DueDateDto, CardDetailPayload }


